### PR TITLE
Add initial io tests and extend measurements

### PR DESCRIPTION
**Pull request by adthoms**
Major 
- Adds SE2 and SE3 tests for pyfg_text.py
- Implement add_pose_landmark_measurement (and associated helper functions/methods) in py_factor_graph/factor_graph.py 
- Allow arbitrary pose-pose, pose-landmark, landmark-landmark range measurement associations in py_factor_graph/measurements.py

Minor
- refactor py_factor_graph/io/pyfg_text.py for code duplication
- refactor py_factor_graph/factor_graph.py for code duplication
- clean up loggers via py_factor_graph/utils/logging_utils.py
- enforce positive qw in function get_quat_from_rotation_matrix in py_factor_graph/utils/matrix_utils.py
- introduce POSE_PRIOR_TYPES and LANDMARK_PRIOR_TYPES in py_factor_graph/priors.py
- bug fix in SE2 rotation precision in py_factor_graph/utils/matrix_utils.py
- simplify covariance getters in classes PosePrior2D and PosePrior3D in py_factor_graph/utils/matrix_utils.py
- enforce pose_landmark naming convention over pose_to_landmark
- add various # TODOs for future development work
- various comment/doc fixes

Let me know your thoughts on these changes.

### CODE REVIEW

File: py_factor_graph/utils/attrib_utils.py [hunk 1]
```diff
@@ -183,7 +183,7 @@ def make_variable_name_validator(type: str) -> Callable:
     form
 
     "<Letter><Number>"
-    Poses should not start with 'L' : Poses = "A1", "B27", "C19"
+    Poses should not start with 'A' : Poses = "A1", "B27", "C19"
```
Comment (alanpapalia) [inline py_factor_graph/utils/attrib_utils.py]: I think this was originally correct?

File: py_factor_graph/utils/attrib_utils.py [hunk 2]
```diff
@@ -228,7 +228,7 @@ def general_variable_name_validator(instance, attribute, value) -> None:
     Return validator for variable names. Should be a string of form
 
     "<Letter><Number>"
-    Poses should not start with 'L' : Poses = "A1", "B27", "C19"
+    Poses should not start with 'A' : Poses = "A1", "B27", "C19"
```
Comment (alanpapalia) [inline py_factor_graph/utils/attrib_utils.py]: Maybe revert to original?

Comment (alanpapalia) [reply py_factor_graph/utils/attrib_utils.py]: We could capitalize the 'NOT' to emphasize this

File: .gitignore
Comment (alanpapalia) [inline .gitignore:1]: Thanks for the cleanup!

File: py_factor_graph/calibrations/range_measurement_calibration.py [hunk 3]
```diff
@@ -6,19 +6,7 @@
...
-    "filename": {"color": "green"},
-    "levelname": {"bold": True, "color": "black"},
-    "name": {"color": "blue"},
-}
-coloredlogs.install(
-    level="INFO",
-    fmt="[%(filename)s:%(lineno)d] %(name)s %(levelname)s - %(message)s",
-    field_styles=field_styles,
-)
+from py_factor_graph.utils.logging_utils import logger
```
Comment (alanpapalia) [inline py_factor_graph/calibrations/range_measurement_calibration.py:9]: ooh very nice!

File: py_factor_graph/utils/matrix_utils.py [hunk 4]
```diff
@@ -142,7 +142,7 @@ def get_info_matrix_from_measurement_precisions(
     assert mat_dim in [3, 6], f"Only support 3x3 or 6x6 info matrices"
     if mat_dim == 3:
         trans_info = [trans_precision] * 2
-        rot_info = [2 * rot_precision]
+        rot_info = [rot_precision]
```
Comment (alanpapalia) [inline py_factor_graph/utils/matrix_utils.py:163]: oh good catch

Comment (alanpapalia) [reply py_factor_graph/utils/matrix_utils.py:163]: Actually, maybe I should look over this? I think your correction may still be right, but there is a conversion from Gaussian to Langevin that may be in play here that makes the 2* correct.

File: py_factor_graph/utils/matrix_utils.py [hunk 5]
```diff
@@ -294,7 +294,7 @@ def get_rotation_matrix_from_quat(quat: np.ndarray) -> np.ndarray:
 
 
 def get_quat_from_rotation_matrix(mat: np.ndarray) -> np.ndarray:
-    """Returns the quaternion from a rotation matrix in scalar-last (x, y, z, w)
+    """Returns the quaternion from a rotation matrix in scalar-last (x, y, z, w). The function ensures w is positive by convention, given R(-q) = R(q)
```
Comment (alanpapalia) [inline py_factor_graph/utils/matrix_utils.py]: lol nice - gotta love double cover

**Review by alanpapalia — COMMENTED** [review]
This all looks great! I need to double check that info matrix function because of the mentioned Gaussian/Langevin conversion. Regardless of what happens I should probably at least add a comment there.

File: py_factor_graph/utils/attrib_utils.py (hunk 1 above)
Comment (adthoms) [reply py_factor_graph/utils/attrib_utils.py]: `not start with` correct, my apologies.

File: py_factor_graph/utils/matrix_utils.py (hunk 4 above)
Comment (adthoms) [reply py_factor_graph/utils/matrix_utils.py:163]: My correction references [SESync_utils.cpp](https://github.com/david-m-rosen/SE-Sync/blob/9b631b6b82d8aa7d32ab846412ae1c070412b7b6/C%2B%2B/SE-Sync/src/SESync_utils.cpp#L77).

Comment (alanpapalia) [reply py_factor_graph/utils/matrix_utils.py:163]: very nice! I'll add a comment and then I think it's good to merge then

**Comment by alanpapalia** [comment]
Just adding this snippet for future instances w.r.t. the 2D Langevin <-> Gaussian case. Kudos to @adthoms for the catch

```python
import numpy as np
from scipy.stats import vonmises, norm
import matplotlib.pyplot as plt

def plot_vonmises_distribution(x, kappa, ax: plt.Axes):
    y = vonmises.pdf(x, kappa)
    ax.plot(x, y, label=r'$\kappa$ = ' + str(kappa), linestyle='-.', color='b')
    ax.legend()

def plot_normal_distribution(x, stddev, ax):
    y = norm.pdf(x, 0, stddev)
    ax.plot(x, y, label='stddev = ' + str(stddev), linestyle='--', color='r')
    ax.legend()

def kappa_to_stddev(kappa):
    return 1 / np.sqrt(kappa)

def kappa_to_covariance(kappa):
    stddev = kappa_to_stddev(kappa)
    return stddev ** 2

if __name__ == '__main__':
    fig, ax = plt.subplots(1, 1)
    kappa = 25
    stddev = kappa_to_stddev(kappa)
    x = np.linspace(-0.3, 0.3, 100)
    plot_vonmises_distribution(x, kappa, ax)
    plot_normal_distribution(x, stddev, ax)
    plt.show()
```

**Comment by alanpapalia** [comment]
@adthoms I think it all looks good now! It'd be great if you want to double check those last commits I just made. Once you're happy I think we're good to merge.

**Review by alanpapalia — APPROVED** [review]
Check my last 3 commits but it LGTM!

**Comment by adthoms** [comment]
@alanpapalia I took the opportunity to do one last clean-up the matrix utils. Several functions were renamed (for consistency) and no breaking changes in pyfg-utilities and range-only-certification are observed. If you are happy, we can merge.

**Pull request merged**